include/drjit/texture.h

```diff
@@ -327,28 +375,32 @@ template <typename _Storage, size_t Dimension> class Texture {
         using Float32 = float32_array_t<Value>;
         using PosF32 = Array<Float32, Dimension>;
 
-        const size_t channels = m_value.shape(Dimension);
         PosF32 pos(pos_);
 
         if constexpr (HasCudaTexture && (sizeof(scalar_t<Value>) <= 4)) {
             if (m_use_accel) {
                 uint32_t pos_idx[Dimension];
                 uint32_t *out_idx =
-                    (uint32_t *) alloca(channels * sizeof(uint32_t));
+                    (uint32_t *) alloca(m_channels_storage * sizeof(uint32_t));
                 for (size_t i = 0; i < Dimension; ++i)
                     pos_idx[i] = pos[i].index();
 
                 // Query coordinates and output values are always single precision
                 jit_cuda_tex_lookup(Dimension, m_handle, pos_idx,
                                     active.index(), out_idx);
 
-                for (size_t ch = 0; ch < channels; ++ch)
-                    out[ch] = Value(Float32::steal(out_idx[ch]));
+                for (size_t ch = 0; ch < m_channels_storage; ++ch) {
+                    Float32 v = Float32::steal(out_idx[ch]);
+
+                    if (ch < m_channels)
+                        out[ch] = Value(v);
+                }
```

Comment:
```suggestion
                for (size_t ch = 0; ch < m_channels; ++ch)
                    out[ch] = Value(Float32::steal(out_idx[ch]));
```

We're just skipping the cases where `ch >= m_channels_storage`, might as well not do them at all, no?

Reply:
I originally had this, but realised we're leaking JIT vars. You still want to steal the variables between `[m_channels, m_channels_storage]` so that we dec ref the underlying JIT variables